game-core/src/main/java/games/strategy/triplea/ui/screen/Tile.java

```diff
@@ -101,10 +101,9 @@ private void draw(final Graphics2D g, final GameData data, final MapData mapData
     // clear
     g.setColor(Color.BLACK);
     g.fill(new Rectangle(0, 0, TileManager.TILE_SIZE, TileManager.TILE_SIZE));
-    for (final List<IDrawable> list : contents.values()) {
-      for (final IDrawable drawable : list) {
-        drawable.draw(bounds, data, g, mapData, unscaled, scaled);
-      }
+    final Queue<IDrawable> queue = new PriorityQueue<>(contents);
```

Comment:
Had to look this up because I wasn't sure if `queue` would have the same ordering as `contents` since no `Comparator` is specified.  Apparently, the `PriorityQueue(Collection)` constructor checks to see if the argument is another `PriorityQueue`, and if so, uses the same `Comparator`, so that the same order is guaranteed.

Reply:
In fact when using the same type of priority queue the underlying array is simply being copied, which makes copying a little bit faster than linear time.